src/Catel.MVVM/MVVM/Providers/WindowLogic.cs

```diff
@@ -183,12 +183,43 @@ public async void OnTargetWindowClosed(object sender, EventArgs e)
                     Log.Warning("Failed to get the 'DialogResult' property of window type '{0}', using 'null' as dialog result", TargetWindow.GetType().Name);
                 }
 
-                await CloseViewModelAsync(dialogResult);
+                await CloseAndDisposeViewModelAsync(dialogResult);
             }
 
             _targetWindowClosedWeakEventListener.Detach();
         }
 
+        /// <summary>
+        /// Closes and disposes the current view model.
+        /// </summary>
+        /// <param name="result"><c>true</c> if the view model should be saved; <c>false</c> if the view model should be canceled; <c>null</c> if it should only be closed.</param>
```

Comment:
What if we put the disposing code into CloseViewModelAsync of LogicBase?

For example, an override like this:

```
public async virtual Task CloseViewModelAsync(bool? result)
{
    // Specify false to keep "backwards compatibility"
    return CloseViewModelAsync(result, false);
}

public async virtual Task CloseViewModelAsync(bool? result, bool dispose)
{
    ... Your code here ...
}
```

Reply:
It's interesting that you brought this up, I was actually thinking about this on the way home from the office yesterday. I agree that this method should be merged into LogicBasic since now both UserControlLogic and WindowLogic are both implementing the same behavior. The original reason why I had not done it was I was trying to keep the amount of changes to a minimum without accounting for the fact "am I duplicating code here?". Tunnel vision is hilarious like that.

I'll task out some time later this evening to make that change and and update the PR.